cypress/integration/plugins/anomaly-detection-dashboards-plugin/create_remote_detector_spec.js

```diff
@@ -0,0 +1,371 @@
+          'content-type': 'application/json',
+          'osd-xsrf': true,
+        },
+        body: remoteClusterSettings,
+      }).then((response) => {
+        Cypress.log({ message: 'Cluster settings updated successfully' });
+        expect(response.status).to.eq(200);
+      });
+
+      cy.wait(5000);
+    });
+
+    // Index some sample data in local and follower cluster (remote)
+    beforeEach(() => {
+      cy.visit(AD_URL.OVERVIEW, { timeout: 10000 });
+      cy.deleteAllIndices();
+      cy.deleteADSystemIndices();
+      cy.wait(3000);
+      const remoteEndpointTestData = `${Cypress.env(
+        'remoteDataSourceNoAuthUrl'
+      )}/${TEST_INDEX_NAME}/_bulk`;
+      const remoteEndpointTestDataTwo = `${Cypress.env(
+        'remoteDataSourceNoAuthUrl'
+      )}/${TEST_SECOND_INDEX_NAME}/_bulk`;
+
+      cy.fixture(AD_FIXTURE_BASE_PATH + 'sample_test_data.txt').then((data) => {
+        cy.request(
+          {
+            method: 'POST',
+            form: false,
+            url: remoteEndpointTestData,
+            headers: {
+              'content-type': 'application/json;charset=UTF-8',
+              'osd-xsrf': true,
+            },
+            body: data,
+          },
+          1000
+        );
+      });
+      cy.fixture(AD_FIXTURE_BASE_PATH + 'sample_remote_test_data.txt').then(
+        (data) => {
+          cy.request(
+            {
+              method: 'POST',
+              form: false,
+              url: remoteEndpointTestDataTwo,
+              headers: {
+                'content-type': 'application/json;charset=UTF-8',
+                'osd-xsrf': true,
+              },
+              body: data,
+            },
+            1000
+          );
+        }
+      );
+      cy.fixture(AD_FIXTURE_BASE_PATH + 'sample_test_data.txt').then((data) => {
+        cy.request({
+          method: 'POST',
+          form: false,
+          url: 'api/console/proxy',
+          headers: {
+            'content-type': 'application/json;charset=UTF-8',
+            'osd-xsrf': true,
+          },
+          qs: {
+            path: `${TEST_INDEX_NAME}/_bulk`,
+            method: 'POST',
+          },
+          body: data,
+        });
+      });
+    });
+
+    it('Full creation - based on remote index', () => {
+      // Define detector step
+      const remoteClusterName = Cypress.env('remoteClusterName');
+
+      cy.visit(AD_URL.CREATE_AD);
+      cy.getElementByTestId('defineOrEditDetectorTitle').should('exist');
+      cy.getElementByTestId('detectorNameTextInput').type(
+        TEST_DETECTOR_REMOTE_NAME
+      );
+      cy.getElementByTestId('detectorDescriptionTextInput').type(
+        TEST_DETECTOR_DESCRIPTION
+      );
+
+      cy.getElementByTestId('clustersFilter').click();
+      cy.contains(
+        '.euiComboBoxOption__content',
+        `${remoteClusterName} (Remote)`
+      ).click();
+
+      cy.wait(500);
+
+      cy.getElementByTestId('indicesFilter').type('sample-ad-index');
```

Comment:
Can we use TEST_INDEX_NAME instead of sample-ad-index ? This applies to all occurrences.

Reply:
changed